linetools/lists/linelist.py

```diff
@@ -595,6 +597,9 @@ def all_transitions(self, line, debug=False):
         elif isinstance(line, Quantity):  # Rest wavelength (with units)
             data = self.__getitem__(line)
             return self.all_transitions(data['name'])
+        elif isinstance(line, tuple): # Zion input
+            data = self.__getitem__(line)
+            return self.all_transitions(data['name'][0].split(' ')[0])
```

Comment:
This seems duplication of code. You could simply replace current line 597: 

`elif isinstance(line, Quantity):` -> `elif isinstance(line, (tuple, Quantity)):`.

Reply:
Fair enough, but then I still have test whether 'line' is a tuple because of the differences between lines 599 and 602.

Reply:
The spliting of name should be done in current line 596...

Reply:
Only if all_transitions returns just 1 transition...

Reply:
Oh, I see. I would prefer avoid duplication of code, so maybe the solution is to have the code check whether the output is `dict` (implying 1 transition) or `Table` (implying >1 transition)